### ISSUE TITLE
User Profile missing change password button for table user

### ISSUE THREAD

**Issue by BrianSChristian** [ctsit/shibboleth_and_table_auth#19]
When patch is applied to REDCap 9.4.0 and user authentication is set to table+Shibboleth, the User Profile page user_profile.php does not show the button for change password for a table user.

**Comment by ChemiKyle** [comment]
Adding this button in reveals that the process of resetting one's own password is not handled identically to the solely table-based process.

I am working on making the actions required to reset a table-based user's password identical for both Shibboleth-table and table settings. So far the method of resetting via email works identically, but users are not prompted to change their password after clicking "Reset password".

**Comment by pbchase** [comment]
@ChemiKyle, the latest fixes you shared with me do not present the security question dialog box after showing the "reset your security question?" dialog:

```
RESET YOUR SECURITY QUESTION?

Do you wish to reset your password security question? If so, you will then 
be redirected back to the My Projects page where you can set up your 
security question again.
```
 Instead, REDCap returns to the "My Projects" page

**Comment by ChemiKyle** [comment]
The inability to "reset your security question" appears to only occur on the server with an actual Shibboleth authentication system set up, in my Docker environment the expected behavior is observed.

**Comment by pbchase** [comment]
When switched out shib_table_auth test system to table-based auth, the "Set Up Password Recovery Question" dialog _will_ appear as it should in both table and shib_table auth.

**Comment by ChemiKyle** [comment]
This is now solved in part of my larger process to refactor REDCap core's table-based authentication. Making these changes into a universal distributable patch will pose a bit more trouble as evidenced by my findings with 9.4.x; however, Rob Taylor's merging of REDCap latest recently may solve the 9.5.x issues in #21

**Comment by pbchase** [comment]
If we have to say some versions of REDCap are unsupportable to make progress I am OK with that. A functional patch a secondary goal to getting this changed merged into REDCap core.

**Comment by ChemiKyle** [comment]
Addressed in PR #22
